src/OpenColorIO/fileformats/FileFormatDiscreet1DL.cpp

```diff
@@ -807,7 +807,8 @@ void TestToolsStripBlank(
     const std::string & stringResult)
 {
     char stringToStrip[200];
-    stringCopy(stringToStrip, stringToStripChar, 200);
+    stringCopy(stringToStrip, stringToStripChar, 199);
+    stringToStrip[199] = '\0';
```

Comment:
I'm wondering if the NULL termination should be forced into the stringCopy macro, or if we should evaluate if a safe version of the copy function could be used directly (do all the platforms support one?)

Reply:
@matthewdcong The GCC9 found a valid 'Linux' issue where the method (a define declared from line 799 to 803) only truncates the string i.e. without adding the null terminated character. 

And @KevinJW is right. The method design is 'weak' as demonstrated by the existing code, and a platform agnostic method must be used.

The solution is to use a **'snprintf'** (i.e. a platform agnostic method) which adds a null termination. Most of the time, the code could be:
`snprintf(dest_buff, sizeof(dest_buff), "%s", src_string);`
( or `sprintf(dest_buff, "%.*s", sizeof(dest_buff) - 1, src_string);`  with a precision ). 

As the destination buffer size is known at compile time, the code could be:
```
constexpr const size_t dstStringSize = 200;
char dstString[dstStringSize];
snprintf(dstString, dstStringSize, "%s", srcString);
```

Reply:
Agreed, it's a valid Linux issue that would manifest if the length of `stringToStrip` was equal to or greater than 200.

`snprintf` definitely seems like the right approach here -- I'll incorporate that into my changes. Thanks for the feedback!